Objects/typeobject.c

```diff
@@ -5410,6 +5413,21 @@ PyType_Ready(PyTypeObject *type)
         }
     }
 
+    /* If __index__ is defined but not __int__, make it default to __index__.
+       Don't touch __float__ and __complex__ as there could be some loss of
+       precision.
+    */
+    PyObject* index = _PyDict_GetItemIdWithError(type->tp_dict, &PyId___index__);
+    if (PyErr_Occurred()) {
```

Comment:
You only need to check `PyErr_Occurred()` if `index == NULL`.

Reply:
Thanks, I made the change